Cython/Compiler/ExprNodes.py

```diff
@@ -746,19 +746,20 @@ def release_temp_result(self, code):
 
     def make_owned_reference(self, code):
         """
-        If result is a pyobject, make sure we own a reference to it.
+        Make sure we own a reference to result.
         If the result is in a temp, it is already a new reference.
         """
-        if self.type.is_pyobject and not self.result_in_temp():
+        if not self.result_in_temp():
             code.put_incref(self.result(), self.ctype())
 
     def make_owned_memoryviewslice(self, code):
         """
         Make sure we own the reference to this memoryview slice.
         """
+        # TODO ideally this would be shared with "make_owned_reference"
         if not self.result_in_temp():
-            code.put_incref_memoryviewslice(self.result(),
-                                            have_gil=self.in_nogil_context)
+            code.put_incref_memoryviewslice(self.result(), self.type,
+                                            have_gil=True)
```

Comment:
Interesting. This looks like it was previously passing the wrong state to `have_gil`. But then, it's either unused (in which case we shouldn't need to pass `have_gil` at all), or it should still depend on `not self.in_nogil_context`, because this method shouldn't just assume we own the GIL.

Reply:
I think this is a mistake and should be `have_gil=self.in_nogil_context`. Nothing seems to be failing with it changed back (but I'll see when it goes through the whole test-suite).